### PR TITLE
[WebNN EP] Support rest Reduce* ops

### PR DESCRIPTION
**Pull request by Honry**
Add ReduceL1, ReduceL2, ReduceLogSum, ReduceLogSumExp, ReduceMin, ReduceProd, ReduceSum, ReduceSumSquare.

### CODE REVIEW

**Comment by Honry** [comment]
@fdwr, @guschmue, PTAL, thanks!

**Review by guschmue — DISMISSED** [review]

File: onnxruntime/core/providers/webnn/builders/impl/reduction_op_builder.cc [hunk 1]
```diff
@@ -117,6 +133,13 @@ bool ReductionOpBuilder::IsOpSupportedImpl(const InitializedTensorSet& /* initia
   if (!GetShape(*input_defs[0], input_shape, logger))
     return false;
 
+  const auto& op_type = node.OpType();
+  // If the optional input 'axes' is provided, it must be an initializer.
+  if (input_defs.size() > 1 && !Contains(initializers, input_defs[1]->Name())) {
+    LOGS(logger, VERBOSE) << "Input axes of " << op_type << " must be known";
```
Comment (fdwr) [inline onnxruntime/core/providers/webnn/builders/impl/reduction_op_builder.cc:139]: Clearer error message? e.g. Input axes must be a constant

**Review by fdwr — APPROVED** [review]
It's nice to have the rest of these.

File: onnxruntime/core/providers/webnn/builders/impl/reduction_op_builder.cc (hunk 1 above)
Comment (Honry) [reply onnxruntime/core/providers/webnn/builders/impl/reduction_op_builder.cc]: Sure, done.

**Review by guschmue — APPROVED** [review]

**Review by fdwr — APPROVED** [review]
Thanks Wanming 👍. We're getting close to full v1 support. 😲

**Pull request merged**